app/src/androidTest/java/com/github/se/orator/endtoend/EndToEndTest.kt

```diff
@@ -250,5 +356,159 @@ class EndToEndAppTest {
+    composeTestRule.onNodeWithText("Sentiment Analysis: 1.0").assertIsDisplayed()
+
+    `when`(speakingRepository.analysisState)
+        .thenReturn(MutableStateFlow(SpeakingRepository.AnalysisState.FINISHED))
+
+    speakingViewModel =
+        SpeakingViewModel(speakingRepository, apiLinkViewModel, userProfileViewModel)
+
+    composeTestRule.runOnUiThread {
+      navController?.navigate(
+          Screen.SPEAKING) // forcing back to home where the user would be once he finishes creating
+      // his account
+    }
+    composeTestRule.onNodeWithTag("mic_text").assertTextContains("Analysis finished.")
+
+    // Step 9: Offline mode:
+
+    composeTestRule.runOnUiThread {
+      navController?.navigate(
+          Screen.OFFLINE) // forcing back to home where the user would be once he finishes creating
+      // his account
+    }
+    composeTestRule.onNodeWithText("No Internet Connection").assertIsDisplayed()
+    composeTestRule
+        .onNodeWithText(
+            "It seems like you don't have any WiFi connection... You can still practice offline!")
+        .assertIsDisplayed()
+    composeTestRule.onNodeWithText("Practice Offline").assertIsDisplayed()
+    composeTestRule.onNodeWithText("Practice Offline").performClick()
+
+    verify(navigationActions).navigateTo(Screen.OFFLINE_INTERVIEW_MODULE)
+    composeTestRule.onNodeWithTag("company").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("jobInput").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("content").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("doneButton").assertIsDisplayed()
+
+    composeTestRule.onNodeWithText("Go to questions screen").assertIsDisplayed()
+
+    composeTestRule.onNodeWithTag("company").performTextInput("Apple")
+    composeTestRule.onNodeWithTag("jobInput").performTextInput("Engineer")
+
+    composeTestRule.onNodeWithTag("company").assertTextContains("Apple")
+    composeTestRule.onNodeWithTag("jobInput").assertTextContains("Engineer")
+
+    composeTestRule.onNodeWithTag("doneButton").performClick()
+    verify(navigationActions).navigateTo(Screen.PRACTICE_QUESTIONS_SCREEN)
+
+    composeTestRule.onNodeWithTag("OfflinePracticeQuestionsScreen").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("BackButton").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("TitleText").assertIsDisplayed()
+    composeTestRule.onNodeWithText("Choose your practice question").assertIsDisplayed()
+
+    composeTestRule.onNodeWithTag("QuestionCard_0")
+    composeTestRule.onNodeWithTag("QuestionCard_1")
+    composeTestRule.onNodeWithTag("QuestionCard_2")
+    composeTestRule.onNodeWithTag("QuestionCard_3")
+    composeTestRule.onNodeWithTag("QuestionCard_4")
+
+    composeTestRule.onNodeWithTag("QuestionText_0")
+    composeTestRule.onNodeWithTag("QuestionText_1")
+    composeTestRule.onNodeWithTag("QuestionText_2")
+    composeTestRule.onNodeWithTag("QuestionText_3")
+    composeTestRule.onNodeWithTag("QuestionText_4")
```

Comment:
Not sure for what these are used. Maybe an `assertExist()` is missing ?

Reply:
Yes indeed... That's fixed now.